opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java

```diff
@@ -110,6 +116,35 @@ public <C> SpanContext extract(Format<C> format, C carrier) {
 
   @Override
   public void close() {
+    if (isClosed) {
```

Comment:
assuming you even need this check, it looks like a race condition. Even if the field access is atomic, a thread can suspend after checking it but before setting it to true.

Reply:
Its true. 

Better to use an `AtomicBoolean` and `compareAndSet`. In most cases this won't be a problem because calling shutdown multiple times on `SdkTracerProvider` is fine. But other implementations may throw if close is called multiple times.